src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -346,6 +346,10 @@ internal static unsafe ParsingStatus TryParseBigIntegerNumber(ReadOnlySpan<char>
             scoped Span<byte> buffer;
             byte[]? arrayFromPool = null;
 
+            if (value.Length == 0)
+            {
+                goto FailExit;
```

Comment:
I'd just duplicate the return, the `goto` makes the code unnecessarily more complex and can break some heuristics